pkg/resources/metrics.go

```diff
@@ -78,6 +78,114 @@ func CreatePostgresConnectivityAlert(ctx context.Context, client k8sclient.Clien
 	return pr, nil
 }
 
+// CreatePostgresResourceStatusPhasePendingAlert creates a PrometheusRule alert to watch for Postgres CR state
+func CreatePostgresResourceStatusPhasePendingAlert(ctx context.Context, client k8sclient.Client, inst *v1alpha1.RHMI, cr *crov1.Postgres) (*prometheusv1.PrometheusRule, error) {
+	if strings.ToLower(inst.Spec.UseClusterStorage) == "true" {
+		logrus.Info("skipping postgres state alert creation, useClusterStorage is true")
+		return nil, nil
+	}
+	productName := cr.Labels["productName"]
+	postgresCRName := strings.Title(strings.Replace(cr.Name, "postgres-example-rhmi", "", -1))
+	alertName := postgresCRName + "PostgresResourceStatusPhasePending"
+	ruleName := fmt.Sprintf("resource-status-phase-pending-rule-%s", cr.Name)
+	alertExp := intstr.FromString(
+		fmt.Sprintf("absent(%s{exported_namespace='%s',resourceID='%s',productName='%s',statusPhase='complete'} == 1)",
+			croResources.DefaultPostgresStatusMetricName, cr.Namespace, cr.Name, productName),
+	)
+	alertDescription := fmt.Sprintf("The creation of the Postgres instance has take longer that %s. Postgres Custom Resource: %s in namespace %s (strategy: %s) for product: %s", alertFor, cr.Name, cr.Namespace, cr.Status.Strategy, productName)
+	labels := map[string]string{
+		"severity":    "warning",
+		"productName": cr.Labels["productName"],
+	}
+	// create the rule
+	pr, err := reconcilePrometheusRule(ctx, client, ruleName, cr.Namespace, alertName, alertDescription, sopUrlPostgresResourceStatusPhasePending, alertExp, labels)
+	if err != nil {
+		return nil, err
+	}
+	return pr, nil
+}
+
+// CreatePostgresResourceStatusPhaseFailedAlert creates a PrometheusRule alert to watch for Postgres CR state
+func CreatePostgresResourceStatusPhaseFailedAlert(ctx context.Context, client k8sclient.Client, inst *v1alpha1.RHMI, cr *crov1.Postgres) (*prometheusv1.PrometheusRule, error) {
+	if strings.ToLower(inst.Spec.UseClusterStorage) == "true" {
+		logrus.Info("skipping postgres state alert creation, useClusterStorage is true")
+		return nil, nil
+	}
+	productName := cr.Labels["productName"]
+	postgresCRName := strings.Title(strings.Replace(cr.Name, "postgres-example-rhmi", "", -1))
+	alertName := postgresCRName + "PostgresResourceStatusPhaseFailed"
+	ruleName := fmt.Sprintf("resource-status-phase-failed-rule-%s", cr.Name)
+	alertExp := intstr.FromString(
+		fmt.Sprintf("%s{exported_namespace='%s',resourceID='%s',productName='%s',statusPhase='failed'} ",
+			croResources.DefaultPostgresStatusMetricName, cr.Namespace, cr.Name, productName),
+	)
+	alertDescription := fmt.Sprintf("The creation of the Postgres instance has been failing longer that %s. Postgres Custom Resource: %s in namespace %s (strategy: %s) for product: %s", alertFor, cr.Name, cr.Namespace, cr.Status.Strategy, productName)
+	labels := map[string]string{
+		"severity":    "critical",
```

Comment:
@austincunningham  Just checking if there's a SOP lined up for new critical alerts

Reply:
@david-martin That will be completed in https://issues.redhat.com/browse/INTLY-7756

Reply:
Thanks @aidenkeating.
re: scheduling, let's aim to land the sop at the same time the rule rolls out in a release (if not already the plan)

Reply:
@david-martin It's not currently in the sprint, will see if we have capacity to add it .

Reply:
@david-martin  added sop url here https://github.com/RHCloudServices/integreatly-help/pull/522